internal/catalogmetadata/client/client.go

```diff
@@ -12,15 +14,57 @@ import (
 	"github.com/operator-framework/operator-controller/internal/catalogmetadata"
 )
 
-func NewClient(cl client.Client) *Client {
-	return &Client{cl: cl}
+// TODO: When a catalogd release containing https://github.com/operator-framework/catalogd/pull/168
+// is made this can be removed in favor of parsing `Catalog.Status` for the
+// appropriate URL for fetching catalog contents.
+const catalogdOnClusterBaseURL = "http://catalogd-catalogserver.catalogd-system.svc"
+const catalogdOnClusterURLTemplate = "%s/catalogs/%s/all.json"
+
+type Opt func(c *Client)
+
+func WithBaseURL(baseURL string) Opt {
+	return func(c *Client) {
+		c.baseURL = baseURL
+	}
+}
+
+func NewClient(cl client.Client, opts ...Opt) *Client {
+	cli := &Client{
+		cl:      cl,
+		baseURL: catalogdOnClusterBaseURL,
+		cache:   map[string]CacheData{},
+	}
+
+	for _, opt := range opts {
+		opt(cli)
+	}
+
+	return cli
+}
+
+type CacheData struct {
+	LastModifiedTime string
+	Bundles          []*catalogmetadata.Bundle
+	Channels         []*catalogmetadata.Channel
 }
 
 // Client is reading catalog metadata
 type Client struct {
 	// Note that eventually we will be reading from catalogd http API
 	// instead of kube API server. We will need to swap this implementation.
 	cl client.Client
+
+	// baseURL is the base URL used by the client when making a request
+	// to the catalogd HTTP server. When using NewClient() this defaults
+	// to http://catalogd-catalogserver.catalogd-system.svc
+	// TODO: When a catalogd release containing https://github.com/operator-framework/catalogd/pull/168
+	// is made this can be removed in favor of parsing `Catalog.Status` for the
+	// appropriate URL for fetching catalog contents.
+	baseURL string
+
+	// cacheInfo is a mapping to hold information necessary
+	// for caching catalog contents
+	cache map[string]CacheData
```

Comment:
what are the keys of this? is it easy to make that obvious in the name of the field?

Reply:
This got moved around but should be addressed. Resolving